core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -531,4 +538,23 @@ protected static Fuzziness randomFuzziness(String fieldName) {
     protected static boolean isNumericFieldName(String fieldName) {
         return INT_FIELD_NAME.equals(fieldName) || DOUBLE_FIELD_NAME.equals(fieldName);
     }
+
+    protected static class MockTermsLookupFetchService extends TermsLookupFetchService {
+
+        private static List<Object> randomTerms = new ArrayList<>();
+
+        public MockTermsLookupFetchService() {
+            super(null, Settings.Builder.EMPTY_SETTINGS);
+            randomTerms.addAll(Lists.newArrayList(generateRandomStringArray(10, 10, false, randomBoolean())));
```

Comment:
yes here allow null means that the returned string array could be null. However, I'm now also "rarely" randomly adding null values as well.